src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -840,18 +844,40 @@ static const char *SDL_GetCPUName(void)
 int SDL_GetCPUCacheLineSize(void)
 {
     const char *cpuType = SDL_GetCPUType();
+    int cacheline_size;
     int a, b, c, d;
     (void)a;
     (void)b;
     (void)c;
     (void)d;
     if (SDL_strcmp(cpuType, "GenuineIntel") == 0 || SDL_strcmp(cpuType, "CentaurHauls") == 0 || SDL_strcmp(cpuType, "  Shanghai  ") == 0) {
         cpuid(0x00000001, a, b, c, d);
-        return ((b >> 8) & 0xff) * 8;
+	cacheline_size = ((b >> 8) & 0xff) * 8;
+        return cacheline_size;
     } else if (SDL_strcmp(cpuType, "AuthenticAMD") == 0 || SDL_strcmp(cpuType, "HygonGenuine") == 0) {
         cpuid(0x80000005, a, b, c, d);
-        return c & 0xff;
+	cacheline_size = c & 0xff;
+        return cacheline_size;
     } else {
+#if defined(__LINUX__) && defined (__GNUC__)
+        if ((cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE)) != -1)
+            return cacheline_size;
+#endif
+#if defined(__LINUX__) /* non glibc linux systems */
+        FILE *f = NULL;
+
+        f = fopen("/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size", "r");
```

Comment:
`f = NULL` and `f = fopen(...)` looks funny, IMO..

Reply:
Should I remove f = NULL? I think compiler would optimize it anyway just wrote it for clarity unless some systems like musl do not return NULL on fopen? I'm not sure if it's a standralized thing because we actually had problems with that in the past on a old machine, but I can remove it?

Reply:
Should I post a commit for __GLIBC__ check or modify f = NULL as well?

Reply:
Why not declare and assign on the same line like `FILE *f = fopen(....);` ?

Reply:
>  some libc libraries like musl do not return NULL on fopen?

Um?? It should return _something_, and that'd be `NULL` in case of failure. Isn't that ANSI C?

Reply:
I had similar problem on an old device where C library didn't return NULL and I had to debug it all the day to see why if wasn't checking for NULL. I don't think that would be a problem on modern glibc or musl or bsd libc. I can replace it if you want to?

Reply:
Well, it can be reorganized like this:

```diff
diff --git a/src/cpuinfo/SDL_cpuinfo.c b/src/cpuinfo/SDL_cpuinfo.c
index 742988c..aad730e 100644
--- a/src/cpuinfo/SDL_cpuinfo.c
+++ b/src/cpuinfo/SDL_cpuinfo.c
@@ -859,19 +859,19 @@ int SDL_GetCPUCacheLineSize(void)
 	cacheline_size = c & 0xff;
         return cacheline_size;
     } else {
-#if defined(__LINUX__) && defined (__GLIBC__)
+#if defined(__LINUX__)
+        FILE *f = NULL;
+#if defined(__GLIBC__)
         if ((cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE)) != -1)
             return cacheline_size;
 #endif
-#if defined(__LINUX__) /* non glibc linux systems */
-        FILE *f = NULL;
-
         f = fopen("/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size", "r");
         if (f) {
             fscanf(f, "%d", &cacheline_size);
             fclose(f);
             return cacheline_size;
 	}
+#endif /* __LINUX__ */
 #elif defined __FreeBSD__
 #ifdef CACHE_LINE_SIZE
         cacheline_size = CACHE_LINE_SIZE;
```

Reply:
Build returns
```
/home/hsnovel/SDL/src/cpuinfo/SDL_cpuinfo.c: In function ‘SDL_GetCPUCacheLineSize_REAL’:
/home/hsnovel/SDL/src/cpuinfo/SDL_cpuinfo.c:877:2: error: #elif without #if
  877 | #elif defined __FreeBSD__
      |  ^~~~
/home/hsnovel/SDL/src/cpuinfo/SDL_cpuinfo.c:882:2: error: #endif without #if
  882 | #endif
      |  ^~~~~
```

I removed the ```#endif /* __LINUX__ */``` so we can do ```#elif defined __FreeBSD__```